src/devTools.js

```diff
@@ -90,12 +128,14 @@ export function send(action, state, options, type, instanceId) {
   }, 0);
 }
 
-export function connect(options = {}) {
+export function connect(options = {}, hostnamePromise) {
   const id = generateId(options.instanceId);
-  start(options);
+  start(options, hostnamePromise);
   return {
     init: (state, action) => {
-      send(action || {}, state, options, 'INIT', id);
+      obtainHostname(hostnamePromise, options, () => {
```

Comment:
The point of making `obtainHostname` an async function was so that we could `await` it here instead of passing it a callback

Reply:
please see [my response here](https://github.com/karol-bisztyga/remotedev/pull/1/files#r607664267)